docs/modules/publisher.md

```diff
@@ -84,29 +75,30 @@ OPC Publisher has several interfaces that can be used to configure it.
 
 ### Configuring Security
 
-IoT Edge provides OPC Publisher with its security configuration for accessing IoT Hub automatically. OPC Publisher can also run as a standalone Docker container by specifying a device connection string for accessing IoT Hub via the `dc` command line parameter. A device for IoT Hub can be created and its connection string retrieved through the Azure Portal.
+IoT Edge provides OPC Publisher with its security configuration for accessing IoT Hub automatically. OPC Publisher can also run as a standalone Docker container by specifying a device connection string for accessing IoT Hub via the `dc` command line parameter. A device for IoT Hub can be created and its connection string retrieved through the Azure portal.
 
-For accessing OPC UA-enabled assets, X.509 certificates and their associated private keys are used by OPC UA. This is called OPC UA application authentication and in addition to OPC UA user authentication. OPC Publisher uses a file system-based certificate store to manage all application certificates. During startup, OPC Publisher checks if there is a certificate it can use in this certificate stores and creates a new self-signed certificate and new associated private key if there is none. Self-signed certificates provide weak authentication, since they are not signed by a trusted Certificate Authority, but at least the communication to the OPC UA-enabled asset can be encrypted this way.
+OPC UA does use X.509 certificates for:
+- mutual authentication of clients and server systems
+- encrypted communication between the systems
+- optionally for user authentication
 
-Security is enabled in the configuration file via the `"UseSecurity": true,` flag. The OPC Publisher will connect to the most secure endpoint available on the OPC UA servers.
-By default, OPC Publisher uses anonymous user authentication (in additional to the application authentication described above). However, OPC Publisher also supports user authentication using username and password. This can be specified via the REST API configuration interface (described below) or the configuration file as follows:
+OPC Publisher can be configured to store these certificates in a file system based certificate store. During startup, OPC Publisher checks if there's already a private certificate it can use. If not, a self-signed certificate is created. Self-signed certificates don't provide any trust value and we don't recommend using them in production.
+
+Encrypted communication can be enabled per endpoint via the `"UseSecurity": true,` flag. By default OPC Publisher will connect to an endpoint using the least secure mode OPC Publisher and the OPC UA server support.
```

Comment:
>least secure mode OPC Publisher and the OPC UA server support [](http://example.com/codeflow?start=152&length=61)

Is this correct or is it really enforcing SecurityPolicy None and if the server does not support it, then the session will fail?

Reply:
@cristipogacean i have updated this as well. pls check.

Reply:
looks good to me!